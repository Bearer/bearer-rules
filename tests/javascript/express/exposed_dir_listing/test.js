const { createInvoker, getEnvironment } = require("../../../helper.js")
const { ruleId, ruleFile, testBase } = getEnvironment(__dirname)

describe(ruleId, () => {
  const invoke = createInvoker(ruleId, ruleFile, testBase)
  
  test("ok_app_use", ({ expect }) => {
    const testCase = "ok_app_use.js"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
  test("serve_index_in_app_use", ({ expect }) => {
    const testCase = "serve_index_in_app_use.js"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
})