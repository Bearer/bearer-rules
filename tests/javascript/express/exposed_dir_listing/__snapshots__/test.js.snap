// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`javascript_express_exposed_dir_listing ok_app_use 1`] = `"{}"`;

exports[`javascript_express_exposed_dir_listing serve_index_in_app_use 1`] = `
"{
  "low": [
    {
      "cwe_ids": [
        "548"
      ],
      "id": "javascript_express_exposed_dir_listing",
      "title": "Missing access restriction on directory listing",
      "description": "## Description\\nInappropriate exposure of a directory listing could give attackers access to sensitive data or source code, either directly or through exploitation of an exposed file structure.\\n\\n## Remediations\\nâœ… Restrict access to sensitive directories and files\\n\\n## Resources\\n- [Express Serve index middleware](https://expressjs.com/en/resources/middleware/serve-index.html)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/javascript_express_exposed_dir_listing",
      "line_number": 9,
      "full_filename": "/tmp/bearer-scan/serve_index_in_app_use.js",
      "filename": ".",
      "source": {
        "start": 9,
        "end": 9,
        "column": {
          "start": 1,
          "end": 52
        }
      },
      "sink": {
        "start": 9,
        "end": 9,
        "column": {
          "start": 1,
          "end": 52
        },
        "content": "app.use(\\"/public\\", serveIndex(__dirname + \\"files\\"))"
      },
      "parent_line_number": 9,
      "snippet": "app.use(\\"/public\\", serveIndex(__dirname + \\"files\\"))",
      "fingerprint": "e5160a3a6c8ab399600f1eef9413fa8d_0",
      "old_fingerprint": "dd4210f84fbf2407460fbaf103e463d2_0",
      "code_extract": "app.use(\\"/public\\", serveIndex(__dirname + \\"files\\"))"
    }
  ]
}"
`;
